.github/workflows/pr_validation.yaml

```diff
@@ -0,0 +1,71 @@
+name: PR checks
+
+concurrency:
+  # Run only for most recent commit in PRs but for all tags and commits on main
+  # Ref: https://docs.github.com/en/actions/using-jobs/using-concurrency
+  group: ${{ github.workflow }}-${{ github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+    branches:
+      - '*'
+  push:
+    branches:
+      - 'main'
+    tags:
+      - '*'
+  workflow_dispatch: {}
+
+jobs:
+  # This job is used to check if the secrets are available. If they are not, we'll skip jobs that require them.
+  should-run-with-secrets:
+    runs-on: ubuntu-latest
+    outputs:
+      result: ${{ steps.check.outputs.result }}
+    steps:
+      - name: Check if secrets are available
+        id: check
+        run: |
+          if [ "${{ secrets.PULP_PASSWORD }}" == "" ]; then
+            echo "result=false" >> $GITHUB_OUTPUT
+          else
+            echo "result=true" >> $GITHUB_OUTPUT
+          fi
+
+  linters:
+    uses: ./.github/workflows/_linters.yaml
+    secrets: inherit
+
+  unit-tests:
+    uses: ./.github/workflows/_unit_tests.yaml
+    secrets: inherit
+
+  integration-tests:
+    needs: should-run-with-secrets
```

Comment:
While making e2e depend on this wholly is fine, with this change for integration tests all outside contributions will not run any integrations tests. Is my understanding correct or have I missed something?

Perhaps we'd rather pass an input to the workflow and use that inside to prevent running enterprise tests?

Reply:
I agree that we might try to limit jobs that require secrets as granularly as possible, but in the end, I think it's not the best way to fix this issue of not running PRs checks for external contributors (that's why I abandoned the idea of making this too complicated and leaking into other workflows). 

IMO we should rather run the whole suite against every PR and we should find a way to do that. @mheap has posted an example of making the PR workflow being triggered on `pull_request_target` instead of `pull_request`: https://gist.github.com/mheap/2c66574509e90747b9fd885e2376935c. I think that ultimately we should adapt to this approach and trigger the PR checks for contributors by adding a label to the PR after inspecting that the changes are safe.

Are you ok with leaving this as is for now and creating a separate issue to tackle the problem "properly"?

Reply:
I'm good with leaving this as is but this topic has been coming back and forth and since this will ultimately break outside contributions (with required checks that we use this will make it not possible to merge PRs from forks without admin override) may I suggest to create an issue to solve this?

Reply:
👍 from me